### PR TITLE
fix: Un-register component keys down the component tree

### PR DESCRIPTION
**Pull request by ufrshubham**
# Description

ComponentKey for children components were not getting unregistered when the sub-tree was removed from the game. As a result:

- same key couldn't be reused
- components with such keys couldn't be readded once removed
- component with such keys were not being garbage collected (didn't actually verify this one though)

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/components/core/component.dart [hunk 1]
```diff
@@ -920,6 +915,15 @@ class Component {
     );
   }
 
+  void _nukeKey() {
```
Comment (spydon) [inline packages/flame/lib/src/components/core/component.dart:918]:
```suggestion
  void _unregisterKey() {
```

**Review by spydon — APPROVED** [review]
Good find!

**Review by luanpotter — APPROVED** [review]
LGTM

File: packages/flame/lib/src/components/core/component.dart (hunk 1 above)
Comment (ufrshubham) [reply packages/flame/lib/src/components/core/component.dart]: Renamed as suggested 👍🏻

**Review by spydon — APPROVED** [review]

**Pull request merged**